dvc/utils/__init__.py

```diff
@@ -371,10 +372,8 @@ def resolve_paths(repo, out):
     dirname = os.path.dirname(abspath)
     base = os.path.basename(os.path.normpath(out))
 
-    # NOTE: `out` might not exist yet, so using `dirname`(aka `wdir`) to check
-    # if it is a local path.
     if (
-        os.path.exists(dirname)  # out might not exist yet, so
+        not urlparse(out).scheme
```

Comment:
`out` may be a local path or a `remote://...` url, but `os.path.exists` isn't a good enough check whether or not `out` is a local path.

In the case where we are importing with `-o subdir/file`, if `subdir` doesn't already exist it is still a valid local output path (and we need to `makedirs(subdir)` later)